Rakefile

```diff
@@ -141,54 +141,91 @@ end
 # PACKAGING ============================================================
 
 if defined?(Gem)
+  GEMS_AND_ROOT_DIRECTORIES = {
+    "sinatra" => ".",
+    "sinatra-contrib" => "./sinatra-contrib",
+    "rack-protection" => "./rack-protection"
+  }
+
   # Load the gemspec using the same limitations as github
-  def spec
+  def spec(gem)
     require 'rubygems' unless defined? Gem::Specification
-    @spec ||= eval(File.read('sinatra.gemspec'))
+    directory = GEMS_AND_ROOT_DIRECTORIES[gem]
+    eval(File.read("#{directory + "/" + gem}.gemspec"))
   end
 
-  def package(ext='')
-    "pkg/sinatra-#{spec.version}" + ext
+  def package(gem, ext='')
+    "pkg/#{gem}-#{spec(gem).version}" + ext
   end
 
-  desc 'Build packages'
-  task :package => %w[.gem .tar.gz].map {|e| package(e)}
+  directory 'pkg/'
+  CLOBBER.include('pkg')
+
+  GEMS_AND_ROOT_DIRECTORIES.each do |gem, directory|
+    file package(gem, '.gem') => ["pkg/", "#{directory + '/' + gem}.gemspec"] do |f|
+      sh "cd #{directory} && gem build #{gem}.gemspec"
+      mv directory + "/" + File.basename(f.name), f.name
+    end
 
-  desc 'Build and install as local gem'
-  task :install => package('.gem') do
-    sh "gem install #{package('.gem')}"
+    file package(gem, '.tar.gz') => ["pkg/"] do |f|
+      sh <<-SH
+        git archive \
+          --prefix=#{gem}-#{source_version}/ \
```

Comment:
@zzak any idea where this `source_version` number comes from? as a CLI argument?